### ISSUE TITLE
Add api Key using environment variables

### ISSUE THREAD

**Issue by bilal-haider-cowlar** [vernemq/vernemq#2196]
### Describe your issue or feature request

I want to add api key when my broker starts using env file. Can you help me with this @ioolkos ?

### Describe the solution you'd like

I want to give api key using docker environment.

### Environment

- VerneMQ Version:
- OS:
- Erlang/OTP version (if building from source):
- Cluster size/standalone:

### Code of Conduct

- [X] I agree to follow the VerneMQ's Code of Conduct

**Comment by ioolkos** [comment]
We have 2 options here:
1) add an API key in the start_script. There's a PR for this here: https://github.com/vernemq/docker-vernemq/pull/197
I think I did not merge it because of the 45 second sleep. But maybe you can find a better way or determine this as a non-issue.

2) We could add the option to set an API key in the VerneMQ schema file. You could then use the normal way to inject ENV variables in Docker. We would have the API key written down in the `vernemq.conf` file.

In short, that's the options to go forward.

The original approach is to use `vmq-admin` to create a key. 

---
👉 Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq
👉 Using the binary VerneMQ packages commercially (.deb/.rpm/Docker) requires a paid subscription.

**Comment by bilal-haider-cowlar** [comment]
So I added these lines in start_vernemq file. 

`if [ -n "$API_KEY" ]; then
  sleep 10 && echo "Adding API_KEY..." && /vernemq/bin/vmq-admin api-key add key="${API_KEY:-DEFAULT}"
  vmq-admin api-key show
fi
wait $pid`

Is there anyway to check that vmq-admin is ready in if statement. So that i can remove sleep. If you can guide me in this I can implement this whole feature and can send you a PR

**Comment by ioolkos** [comment]
I was just asking myself whether we couldn't do this *after* the `wait $pid`. Maybe we can get an exit status and then add a Key depending on the status. Just throwing this out here, it might not work for a thousand reasons.

```
wait $pid
exit_status=$? ...
```

---
👉 Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq
👉 Using the binary VerneMQ packages commercially (.deb/.rpm/Docker) requires a paid subscription.

**Comment by barrettc** [comment]
@ioolkos I tried your suggestion of adding the check for `API_KEY` after the `wait $pid` line and it never executes. I think the reason is that `wait $pid` is waiting for the termination of the vernemq process which won't happen unless vernemq crashes/stops for some reason at which point the `sigterm_handler` would be called. The snippet suggested by @bilal-haider-cowlar works fine for me and the 10 second sleep seems reasonable and would only be invoked if `API_KEY` is passed in.

**Comment by ioolkos** [comment]
@barrettc Thanks for testing!
Okay, if it's wrapped in an "if", I'll accept a PR over on https://github.com/vernemq/docker-vernemq :)
[I can still hate hardcoded delays elsewhere]

---
👉 Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq
👉 Using the binary VerneMQ packages commercially (.deb/.rpm/Docker) requires a paid subscription.

**Comment by barrettc** [comment]
@ioolkos Sounds good - I am going to be out of the office until next week so I'll get a PR together on Monday or Tuesday unless someone beats me to it.

**Comment by ioolkos** [comment]
@barrettc awesome, thanks for your help!

---
👉 Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq
👉 Using the binary VerneMQ packages commercially (.deb/.rpm/Docker) requires a paid subscription.

**Comment by barrettc** [comment]
@ioolkos PR created over at https://github.com/vernemq/docker-vernemq - thanks!

**Comment by barrettc** [comment]
@ioolkos Can you give a sense of when the next release might be coming out with this fix? I didn't see a release cadence schedule listed anywhere so apologies if I missed it
